sources/packages/backend/workflow/test/2021-2022/parttime-assessment/awards-amounts/parttime-assessment-2021-2022-awards-amount-CSGD.e2e-spec.ts

```diff
@@ -0,0 +1,216 @@
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSGDAmount,
+      ),
+    );
+  });
+
+  it("Should determine federalAwardCSGDAmount for 3 or more dependants and calculatedDataTotalFamilyIncome > limitAwardCSGDIncomeCap", async () => {
+    // Arrange
+    const assessmentConsolidatedData =
+      createFakeConsolidatedPartTimeData(PROGRAM_YEAR);
+    assessmentConsolidatedData.studentDataCRAReportedIncome = 42000;
+    assessmentConsolidatedData.studentDataRelationshipStatus = "married";
+    assessmentConsolidatedData.studentDataIsYourSpouseACanadianCitizen =
+      YesNoOptions.Yes;
+    assessmentConsolidatedData.partner1CRAReportedIncome = 42000;
+    assessmentConsolidatedData.studentDataDependants = [
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+      createFakeStudentDependentEligible(
+        DependentEligibility.Eligible0To18YearsOld,
+      ),
+    ];
+    assessmentConsolidatedData.studentDataHasDependents = YesNoOptions.Yes;
+    // Act
+    const calculatedAssessment = await executePartTimeAssessmentForProgramYear(
+      PROGRAM_YEAR,
+      assessmentConsolidatedData,
+    );
+    // Assert
+    // calculatedDataTotalFamilyIncome > limitAwardCSGDIncomeCap
+    // federalAwardCSGDAmount
+    expect(
+      calculatedAssessment.variables.calculatedDataTotalFamilyIncome,
+    ).toBeGreaterThan(
+      calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+        .limitAwardCSGDIncomeCap,
+    );
+    const nestedCalculation = Math.max(
+      calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+        .limitAwardCSGD3OrMoreChildAmount -
+        (calculatedAssessment.variables.calculatedDataTotalFamilyIncome -
+          calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+            .limitAwardCSGDIncomeCap) *
+          calculatedAssessment.variables.dmnPartTimeAwardFamilySizeVariables
+            .limitAwardCSGD3OrMoreChildSlope,
+      0,
+    );
+    const offeringWeeksAmount =
+      Math.round(
+        nestedCalculation *
+          calculatedAssessment.variables.offeringWeeks *
+          10000,
+      ) / 10000;
+    const maxComparisonCalculation = Math.max(
+      offeringWeeksAmount,
+      calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+        .limitAwardCSGD2OrLessChildAmount,
+    );
+    expect(
+      Math.round(
+        calculatedAssessment.variables.federalAwardCSGDAmount * 10000,
+      ) / 10000,
+    ).toBe(
+      Math.min(
+        maxComparisonCalculation,
+        calculatedAssessment.variables.dmnPartTimeAwardAllowableLimits
+          .limitAwardCSGDAmount,
+      ),
+    );
+  });
+
+  it("Should determine federalAwardCSGDAmount for less than 3 dependants and calculatedDataTotalFamilyIncome <= limitAwardCSGDIncomeCap", async () => {
```

Comment:
For the tests testing less than three dependents, should we not consider adding at least one dependent to the `studentDataDependants` in at least one of the scenarios?

Reply:
I didn't add it in the first place because it didn't change the result very much, but I added one dependent for less than 3 dependents scenario in my latest commit to complete the dependents scenarios.